app/controllers/qing_groups_controller.rb

```diff
@@ -0,0 +1,51 @@
+class QingGroupsController < ApplicationController
+  # authorization via cancan
+  load_and_authorize_resource
+
+  before_filter :prepare_qing_group, :only => [:create]
+
+  def new
+    @qing_group = QingGroup.new(:form_id => params[:id])
+    render(:form)
+  end
+
+  def edit
+    render(:form)
+  end
+
+  def show
+    render(:form)
+  end
+
+  def create
+    create_or_update
+  end
+
+  def update
+    @qing_group.assign_attributes(params[:qing_group])
+    create_or_update
+  end
+
+  def destroy
+    destroy_and_handle_errors(@qing_group)
+    redirect_to(edit_form_url(@qing_group.form))
```

Comment:
Since this will be AJAX only, I think we don't need to redirect or use destroy_or_handle_errors. How about this:

```
begin
  @qing_group.destroy
  render nothing: true, status: 204
end
```

If there is an error, Rails will render some junk with status 500, and the AJAX call can catch the 500 and say server error.